server/planning/validate/planning_validate.py

```diff
@@ -116,72 +89,80 @@ def _validate_vocabularies(self, vocabularies, field, value):
         pass
 
 
-class PlanningValidateResource(Resource):
-    endpoint_name = "planning_validator"
-    schema = {
-        "validate_on_post": {"type": "boolean", "default": False},
-        "type": {"type": "string", "required": True},
-        "validate": {"type": "dict", "required": True},
-    }
+def get_validator_schema(schema):
+    """Get schema for given data that will work with validator.
+
+    - if field is required without minlength set make sure it's not empty
+    - if there are keys with None value - remove them
+
+    :param schema
+    """
+    validator_schema = {key: val for key, val in schema.items() if val is not None}
+
+    if validator_schema.get("required") and not validator_schema.get("minlength"):
+        validator_schema.setdefault("empty", False)
+
+    return validator_schema
 
-    resource_methods = ["POST"]
-    item_methods: List[str] = []
 
+def get_filtered_validator_schema(validator, validate_on_post: bool):
+    """Get schema for a given validator, excluding fields with None values,
+    and only include fields that are in enabled_fields."""
 
-class PlanningValidateService(Service):
-    def create(self, docs, **kwargs):
-        for doc in docs:
-            test_doc = deepcopy(doc)
-            doc["errors"] = self._validate(test_doc)
+    enabled_fields = get_enabled_fields(validator)
+    return {
+        field: get_validator_schema(field_schema)
+        for field, field_schema in validator["schema"].items()
+        if field in enabled_fields and field_schema and field_schema.get("validate_on_post", False) == validate_on_post
+    }
+
+
+async def get_validator(doc: dict[str, Any]):
+    """Get validators from planning types service."""
+    return get_resource_service("planning_types").find_one(req=None, name=doc[ITEM_TYPE])
 
-        return [doc["errors"] for doc in docs]
 
-    def _get_validator(self, doc):
-        """Get validators."""
-        return get_resource_service("planning_types").find_one(req=None, name=doc[ITEM_TYPE])
+async def validate_doc(doc: dict[str, Any]):
```

Comment:
Instead of passing in a dictionary of arguments (like we did with the old system), what if we expand the required args out into this function.

This might be:
```python
async def validate_doc(
    item: dict,
    item_type: str,
    validate_on_post: bool = False
) -> list: ...
```
Then we can pass in only what's required. It might help make other functions cleaner as well, as you can then pass in item_type to functions instead of the entire dict.